File: src/bot/Bot.py
from src.bot.Commands import Commands
from src.command.Idle import Idle
from src.utils.Pathfinder import Pathfinder


class Bot:

    def __init__(self):
        self.player_id = None
        self.game_state = None
        self.character_state = None
        self.other_bots = None
        self.commands = None
        self.pathfinder = Pathfinder()

    def set_player_id(self, player_id):
        self.player_id = player_id
        self.commands = Commands(player_id)

    def get_name(self):
        raise NotImplementedError

    def turn(self, game_state, character_state, other_bots):
        self.game_state = game_state
        self.character_state = character_state
        self.other_bots = other_bots
        self.pathfinder.set_game_state(game_state, other_bots)
        return Idle(self.player_id)

File: java/src/Bot.java
import utils.Character;
import utils.Pathfinder;

import java.util.ArrayList;
import java.util.HashMap;

public class Bot {

    protected int player_id;
    protected Command command;
    protected Pathfinder pathfinder;
    protected Character player;
    protected ArrayList<Character> otherBots;

    public Bot(){
        player_id = -1;
        pathfinder = new Pathfinder();
        command = null;
    }

    public void set_player_id(int id){
        player_id = id;
        command = new Command(player_id);
    }

    public HashMap<String, String> turn(String game_state, HashMap<String, Object> character_state, ArrayList<HashMap<String, Object>> other_bots){
        this.player = Character.parse(character_state);
        this.otherBots = new ArrayList<>();
        for (int i = 0; i < other_bots.size(); i++){
            this.otherBots.add(Character.parse(other_bots.get(i)));
        }
        pathfinder.createGraph(game_state, this.otherBots);
        return command.idle();
    }
}
